Sources/TSCBasic/FileSystem.swift

```diff
@@ -485,6 +495,7 @@ public class InMemoryFileSystem: FileSystem {
     private enum NodeContents {
         case file(ByteString)
         case directory(DirectoryContents)
+        case symlink(AbsolutePath)
```

Comment:
I realize that supporting relative symlinks makes this a bit more tricky.  It might be best here to store a `String` with the actual symlink contents and then convert it to a path when used.  There should be an `AbsolutePath` initializer that takes a base path and a string representation of a path, and handles the case of both relative and an absolute string representation (using the base path only if the string representation is relative).